BitPayLib/class-bitpayipnprocess.php

```diff
@@ -262,8 +262,14 @@ private function process_refunded( Invoice $bitpay_invoice, WC_Order $order ): v
 
 	private function process_processing( Invoice $bitpay_invoice, WC_Order $order ): void {
 		$this->validate_bitpay_status_in_available_statuses( $bitpay_invoice, array( 'paid' ) );
-
 		$order->add_order_note( $this->get_start_order_note( $bitpay_invoice->getId() ) . 'is processing.' );
-		$order->update_status( 'processing', __( 'BitPay payment processing', 'woocommerce' ) );
+
+		$wordpress_order_status = $this->get_gateway_settings()['bitpay_checkout_order_process_paid_status'];
+		if ( WcGatewayBitpay::IGNORE_STATUS_VALUE === $wordpress_order_status ) {
```

Comment:
@farjad6 Over on the issue I created bobbrodie mentioned "best for a BitPay Paid Invoice Status setting, with a default matching today so we maintain backwards compatibility"... would this do that? Imagining a merchant who already had this plugin, then they get this update but they haven't yet gone into the settings page and updated... would it go to `wc-processing` status? It doesn't look like it to me, but just asking.

Reply:
@shaunek-hero, yes sir. 
In previous code it was hardcoded to "processing". I am doing the same thing here: 
$new_status = $this->get_wc_order_statuses()[ $wordpress_order_status ] ?? 'processing';
That means if nothing is set take "processing".